submissions/expero1/oop-exercise/index.js

```diff
@@ -0,0 +1,105 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Habitat {
+  static SPECIES = { DOG: "dog", CAT: "cat", HUMAN: "human" };
+  static GENDER = { MALE: "male", FEMALE: "female" };
+  friends = [];
+  PROPS = ["species", "name", "gender", "legs", "saying"];
+  constructor(species, name, gender, legs, saying) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.saying = saying;
+  }
+
+  addFriend(habitat) {
+    if (!this.friends.includes(habitat)) {
+      this.friends.push(habitat);
+    }
+    if (!habitat.friends.includes(this)) {
+      habitat.friends.push(this);
+    }
+  }
+
+  toString() {
+    let propsKeys = this.PROPS.map((key) => {
+      return this[key];
+    });
```

Comment:
We do not need curly braces and `return` keyword when arrow function body has the return statement only.
```
.map((v) => {
  return a[v];
})

.map((v) => a[v])

.map(v => a[v]) // parentheses around arrow function params can also be omitted when parameters count is exactly and no destructuring involved. Also depends on teams' code style conventions

```

Reply:
Your truth. I must use the simplified syntax as it is easier to read. I try to accustom myself to it, but I do not always notice cases when it can be used.